### PR TITLE
Adding reverser for log-group matching struct.

### PR DESCRIPTION
**Pull request by Mercurrent**
Preparation for logical switch at submission stage.

### CODE REVIEW

File: ctpolicy/ctpolicy.go [hunk 1]
```diff
@@ -76,3 +81,17 @@ func lifetimeInMonths(cert *x509.Certificate) int {
 	}
 	return lifetimeInMonths
 }
+
+// GroupByLogs reverses match-map between Logs and Groups. Returns map from log-URLs to set of Group-names.
```
Comment (daviddrysdale) [inline ctpolicy/ctpolicy.go]: nit: for total clarity "Returns map from log-URLs to set of Group-names that contain the log"

File: ctpolicy/ctpolicy_test.go [hunk 2]
```diff
@@ -102,3 +103,113 @@ func TestLifetimeInMonths(t *testing.T) {
...
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := GroupByLogs(test.logGroups)
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("Unable to map logs to groups properly, got \n%v\n, want \n%v\n", got, test.want)
```
Comment (daviddrysdale) [inline ctpolicy/ctpolicy_test.go]: style: usual form is to echo the name of the function under test:
```golang
t.Errorf("GroupByLogs()=%v, want %v", got, test.want)
```

File: ctpolicy/ctpolicy_test.go [hunk 3]
```diff
@@ -102,3 +103,113 @@ func TestLifetimeInMonths(t *testing.T) {
...
+		logGroups map[string]*LogGroupInfo
+		want      map[string]map[string]bool
+	}{
+		{
+			name: "BaseGroup",
+			logGroups: map[string]*LogGroupInfo{
+				BaseName: {
+					name: BaseName,
+					LogURLs: map[string]bool{
+						"ct.googleapis.com/aviator/":   true,
```
Comment (daviddrysdale) [inline ctpolicy/ctpolicy_test.go]: Aside: it makes me wince every time I see these kind of values (non-URLs stored in a field named LogURL...)

Comment (Mercurrent) [reply ctpolicy/ctpolicy_test.go]: Do you prefer URI?

Comment (daviddrysdale) [reply ctpolicy/ctpolicy_test.go]: I prefer URLs (or URIs) to actually be URLs (i.e. to have the protocol identifier)

Comment (Mercurrent) [reply ctpolicy/ctpolicy_test.go]: I can change it for the test (if that's what you want), but not at the real log-lists. Passed your comment to Rob.

File: ctpolicy/ctpolicy_test.go (hunk 2 above)
Comment (Mercurrent) [reply ctpolicy/ctpolicy_test.go]: Changed.

File: ctpolicy/ctpolicy.go (hunk 1 above)
Comment (Mercurrent) [reply ctpolicy/ctpolicy.go]: Done.

**Pull request merged**